pylops/LinearOperator.py

```diff
@@ -47,32 +48,211 @@ class LinearOperator(spLinearOperator):
         .. versionadded:: 2.0.0
 
         Name of operator (to be used by :func:`pylops.utils.describe.describe`)
 
     """
 
-    def __init__(self, Op=None, explicit=False, clinear=None, name=None):
-        self.explicit = explicit
+    def __init__(
+        self,
+        Op=None,
+        dtype=None,
+        shape=None,
+        dims=None,
+        dimsd=None,
+        clinear=None,
+        explicit=None,
+        name=None,
+    ):
         if Op is not None:
             self.Op = Op
-            self.shape = self.Op.shape
-            self.dtype = self.Op.dtype
-            self.clinear = getattr(self.Op, "clinear", True)
+            # All Operators must have shape and dtype
+            dtype = self.Op.dtype if dtype is None else dtype
+            shape = self.Op.shape if shape is None else shape
+            # Optionally, some operators have other attributes
+            dims = getattr(Op, "dims", (Op.shape[1],)) if dims is None else dims
+            dimsd = getattr(Op, "dimsd", (Op.shape[0],)) if dimsd is None else dimsd
+            clinear = getattr(Op, "clinear", True) if clinear is None else clinear
+            explicit = (
+                getattr(self.Op, "explicit", False) if explicit is None else explicit
+            )
+            if explicit and hasattr(Op, "A"):
+                self.A = Op.A
+            name = getattr(Op, "name", None) if name is None else name
+
+        if dtype is not None:
+            self.dtype = dtype
+        if shape is not None:
+            self.shape = shape
+        if dims is not None:
+            self.dims = dims
+        if dimsd is not None:
+            self.dimsd = dimsd
         if clinear is not None:
             self.clinear = clinear
+        if explicit is not None:
+            self.explicit = explicit
         self.name = name
 
+    @property
+    def shape(self):
+        _shape = getattr(self, "_shape", None)
+        if _shape is None:  # Cannot find shape, falling back on dims and dimsd
+            dims = getattr(self, "_dims", None)
+            dimsd = getattr(self, "_dimsd", None)
+            if dims is None or dimsd is None:  # Cannot find both dims and dimsd, error
+                raise AttributeError(
+                    (
+                        f"'{self.__class__.__name__}' object has no attribute 'shape' "
+                        "nor both fallback attributes ('dims', 'dimsd')"
+                    )
+                )
+            _shape = (np.prod(dimsd), np.prod(dims))
+            self._shape = _shape  # Update to not redo everything above on next call
+        return _shape
+
+    @shape.setter
+    def shape(self, new_shape):
+        new_shape = tuple(new_shape)
+        if not isshape(new_shape):
+            raise ValueError(f"invalid shape %{new_shape:r} (must be 2-d)")
+        dims = getattr(self, "_dims", None)
+        dimsd = getattr(self, "_dimsd", None)
+        if dims is not None and dimsd is not None:  # Found dims and dimsd
```

Comment:
is there a case where setting a new shape will not make it incompatible with dims/dimsd?

If not, it sounds like changing shape is never a good idea and should be forbidden maybe?

Reply:
This is also a good point. Currently you can change a shape as long as you don't have dims/dimsd. Maybe this doesn't make sense.

Reply:
Ok, I see. Once we have all operators with dims/dimsd in place then probably this will never be the case but for now we can allow it :)